pkg/http/http.go

```diff
@@ -29,26 +35,97 @@ type ClientOptions struct {
+
+	bodyBuffer := &bytes.Buffer{}
+	bodyWriter := multipart.NewWriter(bodyBuffer)
+
+	fileWriter, err := bodyWriter.CreateFormFile(fieldName, file)
+	if err != nil {
+		return &http.Response{}, errors.Wrapf(err, "error creating form file %v for field %v", file, fieldName)
+	}
+
+	fileHandle, err := os.Open(file)
+	if err != nil {
+		return &http.Response{}, errors.Wrapf(err, "unable to locate file %v", file)
+	}
+	defer fileHandle.Close()
+
+	_, err = io.Copy(fileWriter, fileHandle)
+	if err != nil {
+		return &http.Response{}, errors.Wrapf(err, "unable to copy file content of %v into request body", file)
+	}
+	err = bodyWriter.Close()
+
+	method := http.MethodPost
+	request, err := c.createRequest(method, url, bodyBuffer, header, nil)
+	if err != nil {
+		return &http.Response{}, errors.Wrapf(err, "error creating %v request to %v", method, url)
+	}
+
+	startBoundary := strings.Index(bodyWriter.FormDataContentType(), "=") + 1
+	boundary := bodyWriter.FormDataContentType()[startBoundary:]
+
+	request.Header.Add("Content-Type", "multipart/form-data; boundary=\""+boundary+"\"")
+	request.Header.Add("Connection", "Keep-Alive")
+
+	response, err := httpClient.Do(request)
+	if err != nil {
+		return response, errors.Wrapf(err, "HTTP %v request to %v failed with error", method, url)
+	}
+
+	return c.handleResponse(response)
+}
+
 // SendRequest sends an http request with a defined method
 func (c *Client) SendRequest(method, url string, body io.Reader, header http.Header, cookies []*http.Cookie) (*http.Response, error) {
+	httpClient := c.initialize()
+
+	request, err := c.createRequest(method, url, body, header, cookies)
+	if err != nil {
+		return &http.Response{}, errors.Wrapf(err, "error creating %v request to %v", method, url)
+	}
+	
+	response, err := httpClient.Do(request)
+	if err != nil {
+		return response, errors.Wrapf(err, "error opening %v", url)
+	}
 
-	logger := log.Entry().WithField("package", "SAP/jenkins-library/pkg/http")
+	return c.handleResponse(response)
+}
+
+// SetOptions sets options used for the http client
+func (c *Client) SetOptions(options ClientOptions) {
+	c.timeout = options.Timeout
+	c.username = options.Username
+	c.password = options.Password
+	c.token = options.Token
+	c.logger = log.Entry().WithField("package", "SAP/jenkins-library/pkg/http")
+}
 
+func (c *Client) initialize() (*http.Client) {
 	c.applyDefaults()
 
 	var httpClient = &http.Client{
 		Timeout: c.timeout,
 	}
 
-	logger.Debugf("Timeout set to %v", c.timeout)
+	c.logger.Debugf("Timeout set to %v", c.timeout)
+
+	return httpClient
+}
 
+func (c *Client) createRequest(method, url string, body io.Reader, header http.Header, cookies []*http.Cookie) (*http.Request, error) {
+	c.logger.Debugf("New %v request to %v", method, url)
 	request, err := http.NewRequest(method, url, body)
-	logger.Debugf("New %v request to %v", method, url)
 	if err != nil {
-		return &http.Response{}, errors.Wrapf(err, "error creating %v request to %v", method, url)
```

Comment:
any reason for removing the error details?

Reply:
Yes the same details are wrapped one level higher so it was kind of "doppelt gemoppelt".

Reply:
ok, overlooked that ...